drivers/gpio/gpio_mcux_igpio.c

```diff
@@ -121,6 +121,20 @@ static int mcux_igpio_pin_interrupt_configure(struct device *dev,
 	u8_t icr;
 	int shift;
 
+	if (mode == GPIO_INT_MODE_DISABLED) {
+		key = irq_lock();
+
+		WRITE_BIT(base->ISR, pin, 1);
+		WRITE_BIT(base->IMR, pin, 0);
+		WRITE_BIT(data->pin_callback_enables, pin, 0);
+
+		irq_unlock(key);
+
+		return 0;
+	}
+
+	irq_unlock(key);
```

Comment:
This line is clearly wrong, since `key` hasn't been initialized at this point.

For the change as a whole: I'm not sure it's correct.

If the write to ICR should be skipped when disabling interrupts then the whole icr calculation region should be under a conditional.

Another difference when mode is `GPIO_INT_MODE_DISABLED` is that the existing code writes 0 to the ISR bit while the new code writes 1.  I suspect 1 is the correct value to write in this case (to clear the status bit), and maybe in all cases, but this needs review from NXP folks.

Reply:
Thanks for your review. Indeed, irq_unlock(key)is superfluous and has been removed.

About change flow:
I think ICR will not take effect after interrupts are disabled. ICR will be reconfigured when interrupts are enabled next time.

when mode is GPIO_INT_MODE_DISABLED is that the existing code has the following problems:
1. Before closing the interrupt (setting IMR), the ICR will be changed to 0 first. If the GPIO is low level at this time, the interrupt will always be triggered. ICR can be written when the interrupt is disabled, but the interrupt must be turned off first.
2. As you can see, ISR will be written as 0, which is not consistent with spec. The description of spec is as follows:
> When the active condition has been detected, the corresponding bit remains set until cleared by software.Status flags are cleared by writing a 1 to the corresponding bit position.

This modification can solve both problems。